### PR TITLE
fixing platform deprecation, newlines, and lager upgrade

### PR DESCRIPTION
**Pull request by v-kat**
I wasn't able to get all the checks passing so let me know if there's any other issues

### CODE REVIEW

**Comment by martinsumner** [comment]
Thanks!  Will take a look at this on Monday.

File: src/riak_core_tcp_mon.erl [hunk 1]
```diff
@@ -458,7 +458,7 @@ ssl_test_() ->
         spawn(fun () ->
             %% server
             {ok, S} = ssl:transport_accept(LS),
-            ok = ssl:ssl_accept(S),
+            ok = ssl:handshake(S),
```
Comment (martinsumner) [inline src/riak_core_tcp_mon.erl]: http://erlang.org/doc/man/ssl.html#handshake-1 - looks like it cannot just return OK, will always return a socket

Comment (v-kat) [reply src/riak_core_tcp_mon.erl]: Thanks. Updated. I might have not run the ssl suite locally :(

File: src/riak_core_gen_server.erl [hunk 2]
```diff
@@ -1015,8 +1015,8 @@ format_status(Opt, StatusData) ->
 		      Name
 	      end,
     Header = lists:concat(["Status for generic server ", NameTag]),
-    Log = sys:get_debug(log, Debug, []),
-    Specfic = 
+    Log = sys:get_log(Debug),
```
Comment (martinsumner) [inline src/riak_core_gen_server.erl]: src/riak_core_gen_server.erl
1018: Call to missing or unexported function sys:get_log/1

Comment (martinsumner) [reply src/riak_core_gen_server.erl]: Dialyzer doesn't like this in OTP 20

**Comment by martinsumner** [comment]
The SSL riak_test tests all pass.  However issue with sys:get_log/1.

File: src/riak_core_tcp_mon.erl [hunk 3]
```diff
@@ -458,8 +458,8 @@ ssl_test_() ->
         spawn(fun () ->
             %% server
             {ok, S} = ssl:transport_accept(LS),
-            ok = ssl:ssl_accept(S),
-            ssl_recv_loop(S)
+            {ok, SslSock} = ssl:handshake(S),
```
Comment (martinsumner) [inline src/riak_core_tcp_mon.erl]: This will need to use `deprecated_21` to switch back to ssl_accept in OTP 20

**Comment by martinsumner** [comment]
To pass `make test`, the branch needs to be updated from basho riak_core `develop-3.0` as this resolves a problem which cancels a number of the tests.

**Comment by martinsumner** [comment]
This looks good now after being updated from upstream.  CI tests pass.

**Review by martinsumner — APPROVED** [review]

**Pull request merged**